Sources/TuistKit/Generator/GeneratorModelLoader.swift

```diff
@@ -296,11 +296,17 @@ extension TuistGenerator.CoreDataModel {
 
 extension TuistGenerator.Headers {
     static func from(manifest: ProjectDescription.Headers, path: AbsolutePath, fileHandler: FileHandling) -> TuistGenerator.Headers {
-        let `public` = manifest.public.map { fileHandler.glob(path, glob: $0) } ?? []
-        let `private` = manifest.private.map { fileHandler.glob(path, glob: $0) } ?? []
-        let project = manifest.project.map { fileHandler.glob(path, glob: $0) } ?? []
+        let `public` = manifest.public.map { headerFiles(path: path, glob: $0, fileHandler: fileHandler) } ?? []
+        let `private` = manifest.private.map { headerFiles(path: path, glob: $0, fileHandler: fileHandler) } ?? []
+        let project = manifest.project.map { headerFiles(path: path, glob: $0, fileHandler: fileHandler) } ?? []
         return Headers(public: `public`, private: `private`, project: project)
     }
+
+    private static func headerFiles(path: AbsolutePath, glob: String, fileHandler: FileHandling) -> [AbsolutePath] {
+        return fileHandler.glob(path, glob: glob).filter {
+            $0.extension == "h"
```

Comment:
😬https://github.com/tuist/xcodeproj/blob/2172162c4a92bd53e1054a15fc16869f6df98487/Sources/xcodeproj/Project/Xcode.swift#L54

Reply:
This is exactly what is needed! thanks for spotting that. Will update this shortly